api/envoy/api/v2/listener/listener.proto

```diff
@@ -215,3 +215,17 @@ message ListenerFilter {
     google.protobuf.Any typed_config = 3;
   }
 }
+
+message UdpListenerFilter {
+  // The name of the udp filter to instantiate. The name must match a supported
+  // filter. There are currently no built-in filters.
+  string name = 1 [(validate.rules).string.min_bytes = 1];
+
+  // Filter specific configuration which depends on the filter being instantiated.
+  // See the supported filters for further documentation.
+  oneof config_type {
+    google.protobuf.Struct config = 2;
```

Comment:
For new things we can remove `config` and the `oneof` and just have `typed_config`.